src/tikray/util/rson.py

```diff
@@ -0,0 +1,49 @@
+import typing as t
+
+import orjson
+import rsonpy
+
+
+class RsonTransformer:
+    """
+    A wrapper around rsonpath, a blazing fast, SIMD-powered JSONPath query engine written in Rust.
+
+    - https://github.com/rsonquery/rsonpath
+    - https://github.com/rsonquery/rsonpy
+
+    The wrapper currently satisfies a few basic test cases of the test suite
+    in `test_moksha.py` (idempotency, simple slicing, and unwrapping),
+    mirroring the corresponding jqlang-based test cases.
+
+    Rsonpath can't do much more by design: It's not suitable to perform advanced
+    restructuring, because JSONPath can't do that.
+    """
+
+    def __init__(self, expression: str, **kwargs):  # kwargs for API consistency with other transformers
+        self.expression = expression
+
+    def transform(self, data: t.Any):
+        """
+        Apply rson transformation.
+        """
+        return self.rson_transform(data, self.expression)
+
+    @staticmethod
+    def rson_transform(data: t.Any, expression: str) -> t.Any:
+        """
+        Invoke the rsonpy module for conducting the transformation, using the `orjson` JSON serializer.
+
+        Because rsonpy currently only provides a string-based interface,
+        the code needs to nest a few encoders and decoders.
+
+        > Rsonpath is made to filter document ahead of parsing/loading. Think of it more like a way to
+        > select a small part of the documents before loading it into memory. It will shine in situations
+        > where you have a JSON file too big to hold in memory but you nevertheless want to load parts of it.
+
+        > Rsonpath does not load anything into memory.
+        > It only works on the raw JSON, not on an in-memory data structure.
+        """
+        try:
+            return next(rsonpy.loads(orjson.dumps(data).decode(), expression, json_loader=orjson.loads))
+        except StopIteration:
+            return None
```

Comment:
Note to self: Better add the explanation to the docstring of the `RsonTransformer` class?